### PR TITLE
gh-117142: ctypes: Clean up `globals-to-fix.tsv`

### PR DESCRIPTION
**Pull request by neonene**
Please merge when you think the tsv file can be updated.

cc @encukou

* Issue: gh-117142

### CODE REVIEW

File: Tools/c-analyzer/cpython/globals-to-fix.tsv [hunk 1]
```diff
@@ -350,37 +350,9 @@ Modules/_testclinic.c	-	TestClass	-
...
-Modules/_ctypes/ctypes.h	-	PyCFuncPtrType_Type	-
-Modules/_ctypes/ctypes.h	-	PyCFuncPtr_Type	-
-Modules/_ctypes/ctypes.h	-	PyCPointerType_Type	-
-Modules/_ctypes/ctypes.h	-	PyCPointer_Type	-
-Modules/_ctypes/ctypes.h	-	PyCSimpleType_Type	-
-Modules/_ctypes/ctypes.h	-	PyCStgDict_Type	-
-Modules/_ctypes/ctypes.h	-	PyCStructType_Type	-
-Modules/_ctypes/ctypes.h	-	PyExc_ArgError	-
 Modules/_ctypes/ctypes.h	-	_ctypes_conversion_encoding	-
 Modules/_ctypes/ctypes.h	-	_ctypes_conversion_errors	-
```
Comment (neonene) [inline Tools/c-analyzer/cpython/globals-to-fix.tsv:354]: Unused?

File: Tools/c-analyzer/cpython/globals-to-fix.tsv [hunk 2]
```diff
@@ -350,37 +350,9 @@ Modules/_testclinic.c	-	TestClass	-
...
-Modules/_ctypes/ctypes.h	-	_ctypes_ptrtype_cache	-
 Modules/_ctypes/ctypes.h	-	basespec_string	-
```
Comment (neonene) [inline Tools/c-analyzer/cpython/globals-to-fix.tsv:355]: Unused?

Comment (neonene) [inline Tools/c-analyzer/cpython/globals-to-fix.tsv]: Unused?

**Comment by encukou** [comment]
I found some more; I hope you don't mind me pushing to the PR directly

**Pull request merged**